Exiled.API/Features/Player.cs

```diff
@@ -1666,18 +1666,14 @@ public void TrySetCustomRoleFriendlyFire(Dictionary<string, Dictionary<RoleTypeI
         /// <summary>
         /// Forces the player to reload their current weapon.
         /// </summary>
-        /// <exception cref="InvalidOperationException">If the item is not a firearm.</exception>
-        public void ReloadWeapon() // TODO: Convert to bool instead of Exception
+        /// <returns>True if succesful, otherwise false.</returns>
+        public bool ReloadWeapon()
         {
-            if (CurrentItem is Firearm firearm)
-            {
-                firearm.Base.AmmoManagerModule.ServerTryReload();
-                Connection.Send(new RequestMessage(firearm.Serial, RequestType.Reload));
-            }
-            else
-            {
-                throw new InvalidOperationException("The player's CurrentItem is not a firearm.");
-            }
+            if (!CurrentItem is Firearm firearm)
+                return false;
+            firearm.Base.AmmoManagerModule.ServerTryReload();
```

Comment:
i think the main logic should be in `firearm.Base.AmmoManagerModule.ServerTryReload();` result, for example when you are try to reload a weapon with full magazine, it should return false, no? for current logic it will return false only if you hold non firearm

Reply:
i disagree

Reply:
I agree :trollface:  
bro that would be stupid